client/reader/user-stream/controller.tsx

```diff
@@ -15,49 +16,69 @@ interface Context {
 
 const analyticsPageTitle = 'Reader';
 
+const isNumeric = ( value: string ): boolean => /^\d+$/.test( value );
+
+async function resolveUserId( userIdentifier: string ): Promise< string > {
+	if ( isNumeric( userIdentifier ) ) {
```

Comment:
Nit: Maybe its better to use JS functions instead of custom one.

```suggestion
	if ( Number.isInteger( Number( userIdentifier ) )  ) {
```